### ISSUE TITLE
Question: Can I build win7-compatible builds?

### ISSUE THREAD

**Issue by googleson78** [kazu-yamamoto/unix-time#57]
I am currently building on Windows 10. The resulting program links against `msvcrt.dll`, and looks for the symbol `_create_locale`, which is missing on Windows 7. I looked around the repo a bit and I found some undefs/defs. Is there any way to tell `ghc` that I don't have `_create_locale`, even while I'm not compiling on Windows 7? (This seems related to #51, but also slightly different since I don't want to compile on Windows 7.)

Sorry if this is the wrong place to ask.

**Comment by googleson78** [comment]
I left only the `#ifdef` branches matching `IS_NT61`, and it works fine.

So the only issue is I don't know how I can tell `stack`/`cabal`/`ghc` to `#def` the correct thing.

I tried putting `- ghc-options -optc-DIS_NT61` in my `package.yaml`/`.cabal`, but it didn't seem to work (even when donwloading `unix-time` locally and putting it in its `.cabal`).

This is obviously not your issue, so you can close it if you want to. (but I would greatly appreciate some tip/hint on this!)

**Comment by jneira** [comment]
Try `cc-options`, cabal would pass them in the configure step, when it is used

**Comment by leohaskell** [comment]
Talking about the same situation (building on Windows 10.1 and running the applcation on Windows 7), I think it would be better to have a cabal flag, which could be easily set like `--flag unix-time:win7-compatible`, or something.

I'm willing to implement such a behaviour, if you're okay with that.

**Comment by kazu-yamamoto** [comment]
Cc: @sighingnow

**Comment by sighingnow** [comment]
Such a flag is fine to me, but please set it as `False` by default since there are not many windows 7 users, IMO.

@leohaskell Would you be willing to submit a pull request for such a flag?

**Comment by leohaskell** [comment]
Yes, I'm willing to submit a PR, though, there are some things, that bother me:
With current master (e06dfbf9e3a7fd8689cee452254b265535d96b12), unix-time can be used on Windows 7 only if it was built on Windows 7, otherwise, the library will fail finding `_create_locale` in *msvcrt.dll*
Using the flag IS_NT61, while building on Windows 10, seems to work, but doesn't on Windows 8, since the build fails with ```unknown symbol `timegm'```

So, to make both cases work, I used the *timegm* implementation provided in [cbits/conv.c](https://github.com/kazu-yamamoto/unix-time/blob/e06dfbf/cbits/conv.c#L94)

Can you think of another way to make this work in all cases? (i.e. running on all versions of Windows, irrespective of the version we were building)
If not, are you fine with a flag `explicit-timegm` which would trigger the above-mentioned behaviour? (set to False by default)
